runtime/src/bank/bank_hash_details.rs

```diff
@@ -23,9 +23,51 @@ use {
 
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub(crate) struct BankHashDetails {
-    /// client version
+    /// The client version
     pub version: String,
+    /// The encoding format for account data buffers
     pub account_data_encoding: String,
+    #[serde(rename = "banks")]
+    pub bank_hash_details: Vec<BankHashSlotDetails>,
+}
+
+impl BankHashDetails {
```

Comment:
I'm generally hesitant to use `Set` or any other specific container name unless the container is purely that. Ie, something like:
```rust
type BankHashDetailsMap = HashMap<Slot, BankHashDetails>
```
In this instance, we have additional contents. And, we are providing (plural) details about each bank, so having `BankHashDetails` with an inner `Vec<BankHashDetail>` doesn't fit either. 

Both you and @seanyoung had suggestions so maybe we should adjust the name. However, I don't love any of the presented alternatives. I'm just now offering my rebuttal so I'll give you and Sean the chance to read/think/respond. But, if we don't agree on something that we all like, I think we should push forward as-is. I'll add some comments in for good measure right now too

This type name will not show up in the produced json, and it is only used within this file. So, it will be very easy to update down the road, and pushing now will allow subsequent work (potentially by both Greg and Sean, and maybe me too) to build on top.

Thoughts?

Reply:
ya totally get that and agree. your update looks good to me.

Reply:
`BankHashDetailsSummary`?

Reply:
> `BankHashDetailsSummary`?

Use this name for the outer, and then use `BankHashSlotDetails` or `BankHashDetails` for the inner ?